### ISSUE TITLE
integrate when some of the cell types show batch effect?

### ISSUE THREAD

**Issue by or-lazarescu** [satijalab/seurat#3696]
greetings Seurat team!
my data set contains 3 sample of the same tissue from 3 different donors. my first goal is to identify cell types and for that I first created 3 different Seurat objects, one for each sample, QC-ed them and then merge them to a single object. when I plotted the UMAP I saw that some cell type displayed batch effect, and some didn't (first pic). 
to try and solve that after QC and merge I used SCTransform with vars.to.regress="orig.ident", but that showed even more batch effect - instead of correcting the effect my UMAP was completely separated by sample (second pic).
my question is - using integrate in this case can help if only some cell types show batch effect? wont I lose a lot of information? can I integrate only a part of my data? and if so - how?

I'm attaching the 2 UMAPs I've created
![allVis_umap_groupBy_sample_batchEffect](https://user-images.githubusercontent.com/60340827/98446277-11bc6d00-2125-11eb-91e3-14977d364f64.png)
![allVis2_umap_groupBy_sample](https://user-images.githubusercontent.com/60340827/98446287-2862c400-2125-11eb-9049-a68082e68483.png)

**Comment by kleurless** [comment]
Have you performed any batch effect correction? You mention that you have 3 donors, in that case you should correct for unwanted donor-variation, technical or biological.

**Comment by or-lazarescu** [comment]
I've tried correcting using the vars.to.regress="orig.ident" option after just merging the objects (second pic in the original post).
Lately I also performed integration according to the standard workflow in the integration vignette - which completely solved any batch effect. Haven't yet done any comparison between the integrated and merged object to check what data I have lost in the process, but I'm still looking for a way to handle the batch effect in a more specific way.

if only some of my cell types show batch effect, does that mean that the batch effect is biological only? since each batch is a different donor.
